cognite/client/_api/data_modeling/instances.py

```diff
@@ -49,13 +51,40 @@
 if TYPE_CHECKING:
     from cognite.client import CogniteClient
 
+_DATA_MODELING_SUPPORTED_FILTERS: frozenset[type[Filter]] = frozenset(
+    {
+        filters.And,
+        filters.Or,
+        filters.Not,
+        filters.In,
+        filters.Equals,
+        filters.Exists,
+        filters.Range,
+        filters.Prefix,
+        filters.ContainsAny,
+        filters.ContainsAll,
+        filters.Nested,
+        filters.HasData,
+        filters.MatchAll,
+        filters.Overlaps,
+    }
+)
+
+
+def _validate_filter(filter: Filter | dict | None) -> None:
+    if filter is None or isinstance(filter, dict):
+        return
+    if not_supported := (filter._involved_filter_types() - _DATA_MODELING_SUPPORTED_FILTERS):
+        names = [f.__name__ for f in not_supported]
+        raise ValueError(f"The filters {names} are not supported for Instances")
```

Comment:
Let's not duplicate this in every module. 
```suggestion
def _validate_filter(filter: Filter | dict | None, supported_filters: set[Filter]) -> None:
    if filter is None or isinstance(filter, dict):
        return
    if not_supported := (filter._involved_filter_types() - supported_filters):
        names = [f.__name__ for f in not_supported]
        raise ValueError(f"The filters {names} are not supported for Instances")
```